pandas-stubs/core/groupby/generic.pyi

```diff
@@ -77,74 +113,78 @@ class SeriesGroupBy(GroupBy, Generic[S1, ByT]):
+        downcast: dict | None | NoDefault = ...,
+    ) -> Series[S1] | None: ...
+    def take(
+        self,
+        indices: TakeIndexer,
+        axis: Axis | NoDefault = ...,
+        **kwargs,
+    ) -> Series[S1]: ...
+    def skew(
         self,
+        axis: Axis | NoDefault = ...,
+        skipna: bool = ...,
         numeric_only: bool = ...,
-        min_count: int = ...,
-        engine=...,
-        engine_kwargs=...,
+        **kwargs,
+    ) -> Series: ...
+    @property
+    def plot(self) -> GroupByPlot[Self]: ...
+    def nlargest(
+        self, n: int = ..., keep: Literal["first", "last", "all"] = ...
     ) -> Series[S1]: ...
-    def prod(self, numeric_only: bool = ..., min_count: int = ...) -> Series[S1]: ...
-    def sem(self, ddof: int = ..., numeric_only: bool = ...) -> Series[float]: ...
-    def std(self, ddof: int = ..., numeric_only: bool = ...) -> Series[float]: ...
-    def var(self, ddof: int = ..., numeric_only: bool = ...) -> Series[float]: ...
-    def tail(self, n: int = ...) -> Series[S1]: ...
-    def unique(self) -> Series: ...
+    def nsmallest(
+        self, n: int = ..., keep: Literal["first", "last", "all"] = ...
+    ) -> Series[S1]: ...
+    def idxmin(self, axis: Axis | NoDefault = ..., skipna: bool = ...) -> Series: ...
+    def idxmax(self, axis: Axis | NoDefault = ..., skipna: bool = ...) -> Series: ...
+    def corr(
+        self,
+        other: Series,
+        method: CorrelationMethod = ...,
+        min_periods: int | None = ...,
+    ) -> Series: ...
+    def cov(
+        self, other: Series, min_periods: int | None = ..., ddof: int | None = ...
+    ) -> Series: ...
+    @property
+    def is_monotonic_increasing(self) -> Series[bool]: ...
+    @property
+    def is_monotonic_decreasing(self) -> Series[bool]: ...
     def hist(
         self,
-        by=...,
+        by: IndexLabel | None = ...,
         ax: PlotAxes | None = ...,
         grid: bool = ...,
         xlabelsize: int | None = ...,
         xrot: float | None = ...,
         ylabelsize: int | None = ...,
         yrot: float | None = ...,
         figsize: tuple[float, float] | None = ...,
-        bins: int | Sequence = ...,
+        bins: int | Sequence[int] = ...,
         backend: str | None = ...,
         legend: bool = ...,
         **kwargs,
-    ) -> AxesSubplot: ...
-    def idxmax(self, axis: Axis = ..., skipna: bool = ...) -> Series: ...
-    def idxmin(self, axis: Axis = ..., skipna: bool = ...) -> Series: ...
-    def __iter__(self) -> Iterator[tuple[ByT, Series[S1]]]: ...
-    def diff(self, periods: int = ..., axis: Axis = ...) -> Series: ...
+    ) -> Series: ...  # Series[Axes] but this is not allowed
+    @property
+    def dtype(self) -> Series: ...
+    def unique(self) -> Series: ...
+    # Overrides that provide more precise return types over the GroupBy class
+    @final  # type: ignore[misc]
+    def __iter__(self) -> Iterator[tuple[ByT, Series[S1]]]: ...  # pyright: ignore
```

Comment:
should put the specific `pyright` thing we are ignoring here.

Reply:
For some reason this pyright error and the one on the last line of the same file so not have a specific error code! I cannot add a specific pyright ignore rule here! This is what pyright emits:
`pandas-stubs/core/groupby/generic.pyi:352:9 - error: Method "__iter__" cannot override final method defined in class "BaseGroupBy"`

Reply:
Created a pyright issue: https://github.com/microsoft/pyright/issues/6985

Reply:
Looks like he fixed it and a new release of pyright will be available, so I recommend updating the pyright version in `pyproject.toml` and then using the correct error code.